### ISSUE TITLE
Some suggestions

### ISSUE THREAD

**Issue by jimmywarting** [hoangvu12/requests-proxy#1]
Hi, nice proxy.
Some few suggestions would be to:
- use ESM syntax and change your package.json to include `type: "module"`
- Use NodeJS new built in fetch API in newer NodeJS versions instead of using axios
- Some requests are also compressed... so when you fetch `abc.com` and it responds with `content-length: x` and `content-encoding: gzip` then content-length will be the size of the compressed response. and what you are giving back to the client will be uncompressed... so the content-length will be invalid. 
  therefore it can be a good idea to fetch the requested proxy url as "raw" meaning that no decompression should be done when you are piping the data back to the client

**Comment by hoangvu12** [comment]
Thank you for the suggestions.

So here is my opinion

1. I'll change to ESM. But I don't really get the reason why I should use ESM over CJS.
2. It is available as a experimental feature in Node v17?l, isn't it?
3. So you want a `raw` query to get the non-compressed response right? Will add.

**Comment by jimmywarting** [comment]
just notice something blocking with raw data and node's own fetch implementation is that you can't fetch raw data...

**Comment by hoangvu12** [comment]
> just notice something blocking with raw data and node's own fetch implementation is that you can't fetch raw data...

So I just implemented your suggestions (except native fetch). You could give it a try and let me know if everything is fine.

**Comment by jimmywarting** [comment]
> 3. So you want a raw query to get the non-compressed response right? Will add.

nah, a raw option could mean that it should not try to decompress the response. and simply pass the compressed data right back to the client. (you should not try to decompress or compress the response at all)
